File: tunnel-common/src/main/java/com/tuuzed/tunnel/common/proto/ProtoMessageEncoder.java
package com.tuuzed.tunnel.common.proto;

import com.tuuzed.tunnel.common.proto.internal.ProtoMessageCodec;
import io.netty.buffer.ByteBuf;
import io.netty.channel.ChannelHandlerContext;
import io.netty.handler.codec.MessageToByteEncoder;

public class ProtoMessageEncoder extends MessageToByteEncoder<ProtoMessage> {
    @Override
    protected void encode(ChannelHandlerContext ctx, ProtoMessage msg, ByteBuf out) throws Exception {
        ProtoMessageCodec.encode(msg, out);
    }
}

File: tunnel2-common/src/main/kotlin/tunnel2/common/proto/ProtoMessageEncoder.kt
package tunnel2.common.proto

import io.netty.buffer.ByteBuf
import io.netty.channel.ChannelHandlerContext
import io.netty.handler.codec.MessageToByteEncoder

class ProtoMessageEncoder : MessageToByteEncoder<ProtoMessage>() {

    override fun encode(ctx: ChannelHandlerContext?, msg: ProtoMessage?, outbuf: ByteBuf?) {
        msg ?: return
        outbuf ?: return
        ProtoMessage.encode(msg, outbuf)
    }
}